mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -139,9 +373,19 @@ void mlir::vector::populateVectorLinearizeTypeConversionsAndLegality(
                       ? typeConverter.isLegal(op)
                       : true);
         }
+        if (isa<vector::ShuffleOp>(op)) {
+          return (isLessThanTargetBitWidth(op, targetBitWidth)
+                      ? (typeConverter.isLegal(op) &&
+                         op->getResult(0)
+                                 .getType()
+                                 .cast<mlir::VectorType>()
+                                 .getRank() == 1)
+                      : true);
+        }
         return std::nullopt;
       });
 
-  patterns.add<LinearizeConstant, LinearizeVectorizable>(
+  patterns.add<LinearizeConstant, LinearizeVectorizable, LinearizeVectorShffle,
+               LinearizeVectorExtract, LinearizeVectorExtractStridedSlice>(
```

Comment:
Could we move this to a different `populate` method? Flattening an elementwise op is potentially a no-op but generating shuffles for extract ops could have a noticeable impact in performance

Reply:
I added a new populate method called `populateVectorLinearizeToShuffleRewritePatterns`. If you have a better name suggestion for this please let me know.